megalista_dataflow/uploaders/google_analytics/google_analytics_4_measurement_protocol.py

```diff
@@ -80,17 +81,19 @@ def do_process(self, batch: Batch):
       app_instance_id = row.get('app_instance_id')
       client_id = row.get('client_id')
       user_id = row.get('user_id')
+      timestamp_micros = row.get('timestamp_micros')
 
       if not self._exactly_one_of(app_instance_id, client_id):
         raise ValueError(
           'GA4 MP should be called either with an app_instance_id (for apps) or a client_id (for web)')
     
       if is_event:
-        params = {k: v for k, v in row.items() if k not in ('name', 'app_instance_id', 'client_id', 'uuid', 'user_id')}
+        event_reserved_keys = self.reserved_keys + ['name']
+        params = {k: v for k, v in row.items() if k not in event_reserved_keys}
         payload['events'] = [{'name': row['name'], 'params': params}]
 
       if is_user_property: 
-        payload['userProperties'] = {k: {'value': v} for k, v in row.items() if k not in ('app_instance_id', 'client_id', 'uuid', 'user_id')}
+        payload['userProperties'] = {k: {'value': v} for k, v in row.items() if k not in self.reserved_keys}
         payload['events'] = {'name': 'user_property_addition_event', 'params': {}}
 
       url_container = [f'{self.API_URL}?api_secret={api_secret}']
```

Comment:
Could we add a check [at line 66] for the api_secret to make sure it's set?

Reply:
Done!

Reply:
Caio, I would like to ask for a favor in this change. Would it be possible to add the removal of null values as well?
Last week, I came into an issue where there were null values in some columns and it led to validation issues on Analytics.

Reply:
Hey sure, I can work on that!

Can you provide a failing example? The idea would be to remove any rows where at least one of the parameters of any event or user property is empty? Or should we look into specific null values?

Since the API is pretty open, it may be hard to implement this in a way that still allows for optional parameters. 

Let me know if this makes sense!

Reply:
what I thought about is ignoring the values that are null, still sending the row but without the values that are null to avoid errors from GA validation, for instance adding the clause `and v is not null` in the filter.

do you think it is a good approach?

Reply:
That's a good one, added the check and corresponding tests